### DISCUSSION TITLE
C support

### DISCUSSION THREAD

**Discussion by KnorrFG** [ki-editor/ki-editor#432] [Q&A] [answered]
Hey, I wanted to try out Ki, so I opened it in a C codebase, and it behaved as if I opened a plain text file. Helix has a C-Tree sitter grammar, so I'd expected Ki to show syntax highlighting, and enable AST-based movements, but neither happened. Also, how would I configure a language server for C?

**Comment by wongjiahau** [comment, marked as answer]
Hi @KnorrFG, thanks for trying out! I've just added a PR here (https://github.com/ki-editor/ki-editor/pull/433/files) for that, but basically, all you have to do is update `shared/src/languages.rs`.

**Reply by KnorrFG** [reply]
Cool ty, is that PR just for demonstration purposes, or are you going to merge it?

**Reply by wongjiahau** [reply]
I'm going to merge it

**Reply by wongjiahau** [reply]
By the way, do you mind if I ask how do you find out Ki?

**Reply by KnorrFG** [reply]
It was mentioned here: <https://lobste.rs/s/elcjqa/missing_text_focused_programming>

**Reply by wongjiahau** [reply]
Thanks!
